spring-cloud-context/src/main/java/org/springframework/cloud/context/named/NamedContextFactory.java

```diff
@@ -67,6 +67,12 @@ public void setConfigurations(List<C> configurations) {
 		return new HashSet<>(contexts.keySet());
 	}
 
+	protected void createAndCacheContexts() {
```

Comment:
For ribbon there would have to be either a caveat that only `@RibbonClient` could be eagerly cached, or provide configuration to enumerate all clients that are to be eagerly created. Feign wouldn't have this problem